webpack/ForemanTasks/Components/TasksDashboard/TasksDashboardHelper.js

```diff
@@ -46,50 +46,23 @@ const queryFromUriQuery = uriQuery => {
   return query;
 };
 
-const uriQueryFromQuery = query => {
-  const uriQuery = {};
-
-  Object.entries(uriToQueryMap).forEach(([uriField, queryField]) => {
-    if (query[queryField]) uriQuery[uriField] = query[queryField];
-  });
-
-  if (uriQuery.time_mode === 'last') {
-    uriQuery.time_mode = 'recent';
-  }
-
-  return uriQuery;
-};
-
 export const getQueryFromUrl = () => {
   const uriQuery = getURIQuery(window.location.href);
 
   return queryFromUriQuery(uriQuery);
 };
 
 export const resolveQuery = query => {
-  const uriQuery = uriQueryFromQuery(query);
-
-  const uri = new URI(window.location.href);
-  const { search } = uri.query(true);
-
-  const data = { search, ...uriQuery, page: 1 };
-  const { $, tfm } = window;
-
-  uri.query(URI.buildQuery(data, true));
-  tfm.tools.showSpinner();
-  $.ajax({
-    type: 'get',
-    url: uri.toString(),
-    success(result) {
-      const res = $(`<div>${result}</div>`);
-
-      $('#tasks-table').html(res.find('#tasks-table'));
-    },
-    error({ statusText }) {
-      $('#tasks-table').html(statusText);
-    },
-    complete(result) {
-      tfm.tools.hideSpinner();
-    },
-  });
+  if (query.mode === 'last') {
+    query.mode = 'recent';
```

Comment:
Changing the object the function received is a bad practice and can lead to issues if the consumer who passes the object to the function would want to continue and use the object. I don't see an issue in this specific case but consider changing to:

```js
const { state, result, mode, time } = query;
const uriQuery = {
...,
time_mode: mode === 'last' ? 'recent' : '<the-other-value>',
...
};
```